src/measure_theory/simple_func_dense.lean

```diff
@@ -168,60 +208,147 @@ begin
+  have h_bound : ∀ n, (λ x, (∥approx_on f hf s y₀ h₀ n x - f x∥₊ : ℝ≥0∞) ^ p.to_real)
+      ≤ᵐ[μ] (λ x, ∥f x - y₀∥₊ ^ p.to_real),
+  { exact λ n, eventually_of_forall
+      (λ x, rpow_le_rpow (coe_mono (nnnorm_approx_on_le hf h₀ x n)) to_real_nonneg) },
+  -- (3) The bounding function `λ x, ∥f x - y₀∥ ^ p.to_real` has finite integral
+  have h_fin :  ∫⁻ (a : β), ∥f a - y₀∥₊ ^ p.to_real ∂μ < ⊤,
+  { exact lintegral_rpow_nnnorm_lt_top_of_snorm_lt_top hp_ne_zero hp_ne_top hi },
+  -- (4) The functions "`p`-th power of distance between `f` and the approximation" tend pointwise
+  -- to zero
+  have h_lim : ∀ᵐ (a : β) ∂μ,
+    tendsto (λ n, (∥approx_on f hf s y₀ h₀ n a - f a∥₊ : ℝ≥0∞) ^ p.to_real) at_top (𝓝 0),
+  { filter_upwards [hμ],
+    intros a ha,
+    have : tendsto (λ n, (approx_on f hf s y₀ h₀ n) a - f a) at_top (𝓝 (f a - f a)),
+    { exact (tendsto_approx_on hf h₀ ha).sub tendsto_const_nhds },
+    convert continuous_rpow_const.continuous_at.tendsto.comp (tendsto_coe.mpr this.nnnorm),
+    simp [zero_rpow_of_pos hp] },
+  -- Then we apply the Dominated Convergence Theorem
+  simpa using tendsto_lintegral_of_dominated_convergence _ hF_meas h_bound h_fin h_lim,
 end
 
+lemma mem_ℒp_approx_on [borel_space E]
+  {f : β → E} {μ : measure β} (fmeas : measurable f)
+  (hf : mem_ℒp f p μ) {s : set E} {y₀ : E} (h₀ : y₀ ∈ s) [separable_space s]
+  (hi₀ : mem_ℒp (λ x, y₀) p μ) (n : ℕ) :
+  mem_ℒp (approx_on f fmeas s y₀ h₀ n) p μ :=
+begin
+  refine ⟨(approx_on f fmeas s y₀ h₀ n).ae_measurable, _⟩,
+  suffices : snorm (λ x, approx_on f fmeas s y₀ h₀ n x - y₀) p μ < ⊤,
+  { have : mem_ℒp (λ x, approx_on f fmeas s y₀ h₀ n x - y₀) p μ :=
+      ⟨(approx_on f fmeas s y₀ h₀ n - const β y₀).ae_measurable, this⟩,
+    convert snorm_add_lt_top this hi₀,
+    ext x,
+    simp },
+  -- We don't necessarily have `mem_ℒp (λ x, f x - y₀) p μ`, because the `ae_measurable` part
+  -- requires `ae_measurable.add`, which requires second-countability
+  have hf' : mem_ℒp (λ x, ∥f x - y₀∥) p μ,
+  { have h_meas : measurable (λ x, ∥f x - y₀∥),
+    { simp only [← dist_eq_norm],
+      exact (continuous_id.dist continuous_const).measurable.comp fmeas },
+    refine ⟨h_meas.ae_measurable, _⟩,
+    rw snorm_norm,
+    convert snorm_add_lt_top hf hi₀.neg,
+    ext x,
+    simp [sub_eq_add_neg] },
+  have : ∀ᵐ x ∂μ, ∥approx_on f fmeas s y₀ h₀ n x - y₀∥ ≤ ∥(∥f x - y₀∥ + ∥f x - y₀∥)∥,
+  { refine eventually_of_forall _,
+    intros x,
+    convert norm_approx_on_y₀_le fmeas h₀ x n,
+    rw [real.norm_eq_abs, abs_of_nonneg],
+    exact add_nonneg (norm_nonneg _) (norm_nonneg _) },
+  calc snorm (λ x, approx_on f fmeas s y₀ h₀ n x - y₀) p μ
+      ≤ snorm (λ x, ∥f x - y₀∥ + ∥f x - y₀∥) p μ : snorm_mono_ae this
+  ... < ⊤ : snorm_add_lt_top hf' hf',
+end
+
+lemma tendsto_approx_on_univ_Lp_snorm [opens_measurable_space E] [second_countable_topology E]
+  {f : β → E} (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ⊤) {μ : measure β} (fmeas : measurable f)
```

Comment:
I don't think you need hp_ne_zero for this one.

Reply:
In fact not in the earlier lemma `tendsto_approx_on_Lp_snorm` either.  Thanks!